cmd/api/src/api/v2/agi.go

```diff
@@ -403,27 +404,52 @@ func parseAGMembersFromNodes(nodes graph.NodeSet, selectors model.AssetGroupSele
 		isCustomMember := false
 		// a member is custom if at least one selector exists for that object ID
 		for _, agSelector := range selectors {
-			if agSelector.Selector == node.Properties.Map[common.ObjectID.String()].(string) {
+			if objectId, ok := node.Properties.Map[common.ObjectID.String()].(string); !ok {
+				log.Warnf("objectid is missing for node %d", node.ID)
+			} else if agSelector.Selector == objectId {
 				isCustomMember = true
 			}
 		}
 
+		var (
+			memberObjectId string
+			memberName     string
+		)
+
+		if objectId, ok := node.Properties.Map[common.ObjectID.String()].(string); !ok {
```

Comment:
So it turns out, we do have a better solution to this and that's why we couldn't find other portions of the code that were doing these unsafe conversions. You can use `node.Properties.Get(common.ObjectID.String()).String()` which will return `(string, error)` and assert on the error and otherwise use the string. This should make all of these a bit cleaner and will mean we don't need to do type casting at the web API level.